marquez_client/utils/request_utils.py

```
@@ -0,0 +1,94 @@
+
+import collections
+import json
+import logging
+from http import HTTPStatus
+
+import requests
+
+
+def put_request(marquez_host, path, args=None, **kwargs):
+    full_path = "http://{0}/{1}".format(marquez_host, path)
+    if args:
+        logging.info("args are: " + json.dumps(args))
+    return requests.put(url=full_path, json=args, timeout=kwargs['timeout'])
+
+
+def post_request(marquez_host, path, payload=None, **kwargs):
+    full_path = "http://{0}/{1}".format(marquez_host, path)
+    return requests.post(
+        url=full_path, json=payload, timeout=kwargs['timeout'])
+
+
+def get_request(marquez_host, path, request_args=None, **kwargs):
+    if request_args and not isinstance(request_args, collections.Mapping):
+        raise Exception("Request argument must be a dictionary")
+    full_path = "http://{0}/{1}".format(marquez_host, path)
+    if request_args:
+        kv = request_args.popitem()
+        key = kv[0]
+        value = kv[1]
+
+        arg_str = "{0}={1}".format(key, value)
+        while request_args:
+            kv = request_args.popitem()
+            key = kv[0]
+            value = kv[1]
+            arg_str += "&{0}={1}".format(key, value)
+        full_path = "{0}?{1}".format(full_path, arg_str)
+    return requests.get(full_path, timeout=kwargs['timeout'])
+
+
+def handle_response(response_function):
+    def wrapped_request(*args, **kwargs):
+        response = response_function(*args, **kwargs)
+        status_code = response.status_code
+        response_text = response.text
+
+        if status_code in [
+            HTTPStatus.OK, HTTPStatus.CREATED, HTTPStatus.NO_CONTENT
+        ]:
+            if response_text:
+                return json.loads(response_text)
+            return
+        exception_str = "{0} - {1}".format(status_code, response_text)
+        logging.error(exception_str)
+        if status_code in [HTTPStatus.NOT_FOUND]:
+            raise ResourceNotFoundException(exception_str)
```

I'm not sure if we want to return an exception when there is a NOT_FOUND.
maybe just an empty payload, or a response object including the status code, idk.

---

+1, not found is not an exceptional case

---

I think a `None` could make more sense just to distinguish from something like an empty list

---

Fixed :) 
Thanks for the suggestions here 👍